qutebrowser/browser/webpage.py

```diff
@@ -158,6 +159,12 @@ def _handle_errorpage(self, info, errpage):
                         if QUrl(elem.attribute('src')) == info.url:
                             elem.setAttribute('style', 'display: none')
                     return False
+            elif (error_str.endswith('Path is a directory') and
+                  info.url.scheme() == 'file'):
```

Comment:
Hmm. I wonder if you should do `if (info.domain, info.error) == (QWebPage.QtNetwork, QNetworkReply.ContentOperationNotPermittedError)` here instead, but that might be too generic...

However, if possible I'd like to handle this outside of the error handling here... Mabye in `acceptNavigationRequest`? Have you looked how other browsers using Qt (Otter Browser, QupZilla, Rekonq) do this?